### PR TITLE
feat: update to core24

### PR DESCRIPTION
**Pull request by lengau**
Also removes the deprecated ruff-lsp

### CODE REVIEW

File: snap/snapcraft.yaml [hunk 1]
```diff
@@ -47,41 +45,16 @@ parts:
...
     source: https://github.com/astral-sh/ruff
     source-type: git
-    source-tag: v$SNAPCRAFT_PROJECT_VERSION
+    source-tag: $SNAPCRAFT_PROJECT_VERSION
     source-depth: 1
     stage:
       - bin/ruff
     organize:
       ruff: bin/ruff
-  ruff-lsp:
```
Comment (soumyaDghosh) [inline snap/snapcraft.yaml:56]: Why are we removing the lsp?

Comment (lengau) [reply snap/snapcraft.yaml:56]: As of ruff 0.5 there's a new lsp built into the command (`ruff server`).

Should I alias to that?

Comment (soumyaDghosh) [reply snap/snapcraft.yaml:56]: If they are binding into it. Then, I think we should let it go.

**Review by soumyaDghosh — APPROVED** [review]
Thanks a lot. Looks good to me

**Pull request merged**